Spigot-Server-Patches/0543-Fix-SPIGOT-5824-Bukkit-world-container-is-not-used.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: William Blake Galbreath <Blake.Galbreath@GMail.com>
+Date: Fri, 10 Jul 2020 13:12:33 -0500
+Subject: [PATCH] Fix SPIGOT-5824 Bukkit world-container is not used
+
+
+diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
+index 2f29d7040d37e8be2ba5f8f634011e069a07db2f..d5ae5c9da76134cfa0df125fb96db67071ed77d2 100644
+--- a/src/main/java/net/minecraft/server/Main.java
++++ b/src/main/java/net/minecraft/server/Main.java
+@@ -86,7 +86,20 @@ public class Main {
+                 return;
+             }
+ 
+-            File file = (File) optionset.valueOf("universe"); // CraftBukkit
++            // Paper start - fix SPIGOT-5824
++            File file;
++            if (optionset.has("universe")) {
++                file = (File) optionset.valueOf("universe"); // CraftBukkit
++            } else {
++                try {
++                    file = new File(org.bukkit.configuration.file.YamlConfiguration
++                            .loadConfiguration((File) optionset.valueOf("bukkit-settings"))
```

Comment:
Use `load()` from YamlConfiguration object.

```java
public static YamlConfiguration loadConfiguration(@NotNull Reader reader) {
    Validate.notNull(reader, "Stream cannot be null");

    YamlConfiguration config = new YamlConfiguration();

    try {
        config.load(reader);
    } catch (IOException ex) {
        Bukkit.getLogger().log(Level.SEVERE, "Cannot load configuration from stream", ex);
    } catch (InvalidConfigurationException ex) {
        Bukkit.getLogger().log(Level.SEVERE, "Cannot load configuration from stream", ex);
    }

    return config;
}
```

When IOException or InvalidConfigurationException occurs it will result in NPE since `Bukkit.getServer()` returns null until CraftServer sets `Bukkit.setServer(...)` in its constructor.

Reply:
Good call ^_^